src/apps/natdiscovery/natdiscovery.c

```diff
@@ -300,8 +300,8 @@ static int run_stunclient_hairpinning(ioa_addr *local_addr, ioa_addr *remote_add
 	if(ret){
 		ret=stunclient_receive(udp_fd2, local_addr, reflexive_addr, other_addr, rfc5780);
 	}
-	close(udp_fd);
-	close(udp_fd2);
+	socket_closesocket(udp_fd);
```

Comment:
This is also somehow gets converted to plain `close()` using macro?
I feel this is an unexpected and undesired side-effect

Reply:
Yeah, the macro is defined src/apps/common/apputils.h and has been since the beginning of time (well, since the code was imported to git).
While both unix socket and winsock opens a socket with socket, winsock closes a socket with closesocket. In what way do you mean it is an undesired side-effect to use socket_closesocket()?

Reply:
Sorry I thought the new macros does that. Disregard my comment then.

Reply:
Ahh, good, then I'm less confused :)
I am aware those expansion macros are kind of a brainfuck to read. But all they do is to create inline functions for checking errno (or WSAGetLastError() in case of Windows) against each error error code.

The fact that errno and for example close() still exists when you build on Windows, but don't apply for sockets, makes things a bit complicated. There is always the risk that a new contributor will add new calls to close() and new checks using errno directly. Everything will compile fine, but not behaving as expected on Windows.